core/lib/zksync_core/src/consensus/mod.rs

```diff
@@ -19,6 +25,71 @@ mod tests;
 
 pub use config::{Config, Secrets};
 
+/// Runs the consensus in the main node mode.
+pub async fn run_main_node(
+    ctx: &ctx::Ctx,
+    cfg: MainNodeConfig,
+    pool: ConnectionPool<Core>,
+    mut stop_receiver: watch::Receiver<bool>,
+) -> anyhow::Result<()> {
+    scope::run!(ctx, |ctx, s| async {
+        s.spawn_bg(async {
+            // Consensus is a new component.
+            // For now in case of error we just log it and allow the server
+            // to continue running.
+            if let Err(err) = cfg.run(ctx, Store(pool)).await {
+                tracing::error!(%err, "Consensus actor failed");
+            } else {
+                tracing::info!("Consensus actor stopped");
+            }
```

Comment:
this doesn't belong to the consensus module, it should be up to the user to decide whether consensus actor failure is fatal.

Reply:
Same reasoning as above: have a single implementation for the time being. User that wants consensus actor failure to be fatal may easily write their own implementation of this function.